js_modules/dagit/packages/core/src/partitions/DimensionRangeWizard.tsx

```diff
@@ -44,6 +45,22 @@ export const DimensionRangeWizard: React.FC<{
 
   const [showCreatePartition, setShowCreatePartition] = React.useState(false);
 
+  const didSetInitialPartition = React.useRef(false);
+  React.useEffect(() => {
+    if (didSetInitialPartition.current) {
+      return;
+    }
+    if (partitionKeys.length) {
+      didSetInitialPartition.current = true;
+    }
+    const query = qs.parse(window.location.search, {ignoreQueryPrefix: true});
+    const partition = query.partition as string;
+    if (partition && partitionKeys.includes(partition)) {
+      setSelected([partition]);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [partitionKeys]);
```

Comment:
Why is the lint override necessary here? I'm not spotting what's missing.

Reply:
oh yeah not necessary anymore. Originally I was leaving partitionKeys out but I have to add it since it has to loads first.

Reply:
oh, it's `setSelected` which unfortunately isn't memoized by the callers so it would cause an infinite loop if included.

Reply:
Ah, I mistook it for a regular `useState` setter.

That's really too bad. The infinite loop issue isn't resolved by the `didSetInitialPartition` ref?

Reply:
I guess it would but we'd still be needlessly entering this useEffect every time the parent component renders